Sources/Sentry/SentryScope.m

```diff
@@ -496,7 +496,8 @@ - (SentryEvent *__nullable)applyToEvent:(SentryEvent *)event
         event.fingerprint = fingerprints;
     }
 
-    if (event.breadcrumbs == nil) {
+    if (event.breadcrumbs == nil
```

Comment:
`m`: I think it would be a bit cleaner if the scope didn't know about replay. In the client, we already check if an event is a replay before we call https://github.com/getsentry/sentry-cocoa/blob/3de9971d99752d282bfbb4b3d977295306275540/Sources/Sentry/SentryClient.m#L624

So, we could remove the breadcrumbs after calling `applyToEvent` in the client. 

It would be great to have some tests for this.